### DISCUSSION TITLE
Set Time Out need for my other secript with ThumbmarkJS?

### DISCUSSION THREAD

**Discussion by LuxLOL** [thumbmarkjs/thumbmarkjs#41] [Q&A]
Hi,
Is it possible to wait till the ThumbmarkJS gets all the data before another script gets executed?
Someone said to me that I need to use the "set time out function" in my other script because the FingerPrintJS will take some micro seconds to identify the browser. 🤔

**Comment by ilkkapeltola** [comment]
this is pretty basic javascript knowledge rather than belonging to this repository.

you can call await in front of a function that returns a promise and it will wait for the function to finish, but that's not really good practice.
you should learn about how promises work and how they are handled when they get resolved.
